### DISCUSSION TITLE
How to store and use an array of values?

### DISCUSSION THREAD

**Discussion by supunmalinga** [usebruno/bruno#426] [Q&A]
Hello, I'm a postman user trying to switch. My current pm test cases have a list of hostnames stored in a datafile where I use that to test a single test case against multiple hosts for API validation. Is there a way to do that with Bruno?. TIA!

**Comment by helloanoop** [comment]
Hey @supunmalinga 

Some questions - so that I can better formulate an answer

1. How many hostnames do you have ?
2. In your whole collection, how many requests do you have ?
3. Currently how are you doing this in postman. Is it by using a collection runner by parsing a csv file ?

**Reply by supunmalinga** [reply]
Hey @helloanoop.
Thanks for the quick response!
1. It varies with the environment but can go up-to about 10-15.
2. Have about 300 requests
3. It's a JSON file, Using the datafile feature in pm -> https://learning.postman.com/docs/collections/running-collections/working-with-data-files/

**Reply by helloanoop** [reply]
So, you have around 300 tests that you need to need to run and each iteration runs all 300 requests against a specific hostname. Is that right?

Also is the hostname the only thing that is stored? Or are there other things that are also stored in your data file (I am assuming CSV)?

**Reply by supunmalinga** [reply]
Yeah pretty much. Hostname is the only variable that changes as of now. I have other variables defined that stay constant (which is easy to do in Bruno as well). File type is actually json. However, doesn't have to be same in Bruno.
eg:
![image](https://github.com/usebruno/bruno/assets/6544722/c6080a8e-e8e3-44f8-8d99-e7c72b81a242)

**Reply by supunmalinga** [reply]
Hello @helloanoop, any tips on how I can make it work ?. Even storing the hostnames within the pre-request scripts as an array might work for this ?.
